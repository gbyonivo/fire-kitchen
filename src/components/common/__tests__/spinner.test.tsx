import { render } from "@testing-library/react";
import { Spinner, SpinnerSize } from "../spinner";

describe("Spinner", () => {
  test.each([[undefined], ["small"], ["medium"], ["large"]])(
    "%s should match snapshot",
    (size) => {
      const { container } = render(<Spinner size={size as SpinnerSize} />);
      expect(container).toMatchSnapshot();
    }
  );

  afterEach(() => {
    jest.clearAllMocks();
  });
});
