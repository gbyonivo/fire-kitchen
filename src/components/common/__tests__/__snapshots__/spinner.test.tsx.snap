// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner large should match snapshot 1`] = `
<div>
  <div>
    <svg
      aria-hidden="true"
      class="animate-spin text-red-500 undefined w-16 h-16"
      data-slot="icon"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.963 2.286a.75.75 0 0 0-1.071-.136 9.742 9.742 0 0 0-3.539 6.176 7.547 7.547 0 0 1-1.705-1.715.75.75 0 0 0-1.152-.082A9 9 0 1 0 15.68 4.534a7.46 7.46 0 0 1-2.717-2.248ZM15.75 14.25a3.75 3.75 0 1 1-7.313-1.172c.628.465 1.35.81 2.133 1a5.99 5.99 0 0 1 1.925-3.546 3.75 3.75 0 0 1 3.255 3.718Z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`Spinner medium should match snapshot 1`] = `
<div>
  <div>
    <svg
      aria-hidden="true"
      class="animate-spin text-red-500 undefined w-10 h-10"
      data-slot="icon"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.963 2.286a.75.75 0 0 0-1.071-.136 9.742 9.742 0 0 0-3.539 6.176 7.547 7.547 0 0 1-1.705-1.715.75.75 0 0 0-1.152-.082A9 9 0 1 0 15.68 4.534a7.46 7.46 0 0 1-2.717-2.248ZM15.75 14.25a3.75 3.75 0 1 1-7.313-1.172c.628.465 1.35.81 2.133 1a5.99 5.99 0 0 1 1.925-3.546 3.75 3.75 0 0 1 3.255 3.718Z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`Spinner small should match snapshot 1`] = `
<div>
  <div>
    <svg
      aria-hidden="true"
      class="animate-spin text-red-500 undefined w-4 h-4"
      data-slot="icon"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.963 2.286a.75.75 0 0 0-1.071-.136 9.742 9.742 0 0 0-3.539 6.176 7.547 7.547 0 0 1-1.705-1.715.75.75 0 0 0-1.152-.082A9 9 0 1 0 15.68 4.534a7.46 7.46 0 0 1-2.717-2.248ZM15.75 14.25a3.75 3.75 0 1 1-7.313-1.172c.628.465 1.35.81 2.133 1a5.99 5.99 0 0 1 1.925-3.546 3.75 3.75 0 0 1 3.255 3.718Z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;

exports[`Spinner undefined should match snapshot 1`] = `
<div>
  <div>
    <svg
      aria-hidden="true"
      class="animate-spin text-red-500 undefined w-10 h-10"
      data-slot="icon"
      fill="currentColor"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M12.963 2.286a.75.75 0 0 0-1.071-.136 9.742 9.742 0 0 0-3.539 6.176 7.547 7.547 0 0 1-1.705-1.715.75.75 0 0 0-1.152-.082A9 9 0 1 0 15.68 4.534a7.46 7.46 0 0 1-2.717-2.248ZM15.75 14.25a3.75 3.75 0 1 1-7.313-1.172c.628.465 1.35.81 2.133 1a5.99 5.99 0 0 1 1.925-3.546 3.75 3.75 0 0 1 3.255 3.718Z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</div>
`;
